github_actions/lib/dependabot/github_actions/file_fetcher.rb

```diff
@@ -39,11 +39,16 @@ def fetch_files
       end
 
       def workflow_files
-        @workflow_files ||=
+        @workflow_files ||= [fetch_file_if_present("action.yml"), fetch_file_if_present("action.yaml")].compact
+
+        # In the special case where the root directory is defined we also scan
+        # the .github/workflows/ folder.
+        return @workflow_files unless directory == "/"
+
+        @workflow_files +=
```

Comment:
I think this will now keep appending these files to the array every time this method is called, where the expected behavior is that all the files are fetched once, and if we have them in memory just return them, what I'd suggest is write something like this:

```ruby
return @workflow_files if defined? @workflow_files

@workflow_files = [fetch_file_if_present("action.yaml") #.. etc]

@workflow_files += ...
```

Reply:
Oh, that is why the ||= operator was there before right?
I have changed it, but I still kept the "unless directory == /" check since we do not want to scan that one in case a specific path is given.